src/server/visual-diff-plugin.js

```diff
@@ -114,56 +139,100 @@ export function visualDiff({ updateGoldens = false, runSubset = false } = {}) {
-					currentRun = session.testRun;
-					clearedDirs.clear();
-				}
+			const screenshotFile = join(newPath, PATHS.FAIL, browser, newName);
+			const screenshotFileName = `${screenshotFile}.png`;
+
+			if (command === 'brightspace-visual-diff-compare') {
+				if (!isCI) { // CI will be a fresh .vdiff folder each time and only one run
+					if (session.testRun !== currentRun) {
+						currentRun = session.testRun;
+						clearedDirs.clear();
+					}
 
-				if (runSubset || currentRun > 0) {
-					if (!clearedDirs.has(newPath)) {
-						clearedDirs.set(newPath, clearDir(updateGoldens, newPath));
+					if (runSubset || currentRun > 0) {
+						if (!clearedDirs.has(newPath)) {
+							clearedDirs.set(newPath, clearDir(updateGoldens, newPath));
+						}
+						await clearedDirs.get(newPath);
 					}
-					await clearedDirs.get(newPath);
 				}
-			}
 
-			const opts = { margin: DEFAULT_MARGIN, ...payload.opts };
-
-			const page = session.browser.getPage(session.id);
-			await page.screenshot({
-				animations: 'disabled',
-				clip: {
-					x: payload.rect.x - opts.margin,
-					y: payload.rect.y - opts.margin,
-					width: payload.rect.width + (opts.margin * 2),
-					height: payload.rect.height + (opts.margin * 2)
-				},
-				path: updateGoldens ? goldenFileName : screenshotFileName
-			});
-
-			if (updateGoldens) {
-				return { pass: true };
-			}
+				const opts = { margin: DEFAULT_MARGIN, ...payload.opts };
+
+				const page = session.browser.getPage(session.id);
+				await page.screenshot({
+					animations: 'disabled',
+					clip: {
+						x: payload.rect.x - opts.margin,
+						y: payload.rect.y - opts.margin,
+						width: payload.rect.width + (opts.margin * 2),
+						height: payload.rect.height + (opts.margin * 2)
+					},
+					path: updateGoldens ? goldenFileName : screenshotFileName
+				});
+
+				if (updateGoldens) {
+					return { pass: true };
+				}
 
-			const goldenExists = await checkFileExists(goldenFileName);
-			if (!goldenExists) {
-				return { pass: false, message: 'No golden exists. Use the "--golden" CLI flag to re-run and re-generate goldens.' };
-			}
+				const goldenExists = await checkFileExists(goldenFileName);
+				if (!goldenExists) {
+					return { pass: false, message: 'No golden exists. Use the "--golden" CLI flag to re-run and re-generate goldens.' };
+				}
 
-			const screenshotInfo = await stat(screenshotFileName);
-			const goldenInfo = await stat(goldenFileName);
+				const screenshotImage = PNG.sync.read(await readFile(screenshotFileName));
+				const goldenImage = PNG.sync.read(await readFile(goldenFileName));
+
+				if (screenshotImage.width === goldenImage.width && screenshotImage.height === goldenImage.height) {
+					const diff = new PNG({ width: screenshotImage.width, height: screenshotImage.height });
+					const pixelsDiff = pixelmatch(
+						screenshotImage.data, goldenImage.data, diff.data, screenshotImage.width, screenshotImage.height, { threshold: DEFAULT_TOLERANCE }
```

Comment:
Can you pass the `diffMask` option so that they're transparent? I've added it my branch but it keeps getting conflicts/blown away each time I rebase.